core/Pos/Util/Util.hs

```diff
@@ -619,3 +624,14 @@ parseJSONWithRead :: Read a => A.Value -> A.Parser a
 parseJSONWithRead =
     either (fail . toString) pure . readEither @String <=<
     parseJSON
+
+----------------------------------------------------------------------------
+-- Exceptions
+----------------------------------------------------------------------------
+
+-- | Catch and log an exception, then rethrow it
+logException :: LoggerName -> IO a -> IO a
+logException name = E.handle (\e -> handler e >> E.throw e)
+  where
+    handler :: E.SomeException -> IO ()
+    handler = usingLoggerName name . logError . show
```

Comment:
It's better to use `displayException` instead of `show`. Also there in `instance Buildable SomeException` in `serokell-util`. You can import `Serokell.Util.Exceptions` and use `pretty` here.